pyrefly/lib/alt/expr.rs

```diff
@@ -705,6 +729,33 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
         }
     }
 
+    /// Check if a protocol is a data protocol (has non-method members)
+    fn is_data_protocol(&self, cls: &Class) -> bool {
+        // A data protocol has at least one non-method member
+        // For simplicity, we consider any annotated field that's not a method to be a data member
```

Comment:
You can get the names of all the protocol members via the protocol metadata (which you can get at using `ClassMetadata::protocol_metadata()`): https://github.com/facebook/pyrefly/blob/6dd81f78c535c9558561010fff75c47b1265fa0a/pyrefly/lib/alt/types/class_metadata.rs#L330.